src/auth/auth.controller.ts

```diff
@@ -63,4 +64,36 @@ export class AuthController {
       };
     }
   }
+
+  @HttpCode(HttpStatus.OK)
+  @Post('register')
+  @UsePipes(new ValidationPipe({ transform: false }))
+  async signUp(@Body() userData: CreateUsersDto) {
+    try {
+      this.logger.log(`Registering user '${userData.username}'...`);
+      const result = await this.authService.signUp(userData);
+
+      if (result.statusCode == 200) {
```

Comment:
if condition ini biso kito bagusin lagi dengan diwrap di fungsi sendiri, biar block try catchnyo biso fokus di catch error `const result = await this.authService.signUp(userData);`